src/NHibernate/Driver/NDataReader.cs

```diff
@@ -549,6 +552,21 @@ internal static NResult Create(DbDataReader reader, bool isMidstream)
 				return result;
 			}
 
+
+			private static DataTable SafeGetSchemaTable(IDataReader reader, out NotSupportedException exception)
+			{
+				exception = null;
+				try
+				{
+					return reader.GetSchemaTable();
+				}
+				catch (NotSupportedException e)
+				{
+					exception = e;
```

Comment:
Adding a `ReflectHelper.PreserveStackTrace(e);` here would be better in my opinion. This way, if re-thrown later, the stack trace will not be overridden. So this would reduce the likeliness of users asking here about this lack of support.